temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -416,40 +407,49 @@ private AttemptTaskHandlerImpl(ActivityTaskHandler handler) {
 
     @Override
     public void handle(LocalActivityAttemptTask attemptTask) throws Exception {
-      attemptTask.markAsTakenFromQueue();
+      SlotReleaseReason reason = SlotReleaseReason.taskComplete();
 
       // cancel scheduleToStart timeout if not already fired
       @Nullable ScheduledFuture<?> scheduleToStartFuture = attemptTask.getScheduleToStartFuture();
       boolean scheduleToStartFired =
           scheduleToStartFuture != null && !scheduleToStartFuture.cancel(false);
 
       LocalActivityExecutionContext executionContext = attemptTask.getExecutionContext();
+      executionContext.newAttempt();
       PollActivityTaskQueueResponseOrBuilder activityTask = attemptTask.getAttemptTask();
 
-      // if an activity was already completed by any mean like scheduleToClose or scheduleToStart,
-      // discard this attempt, this execution is completed.
-      // The scheduleToStartFired check here is a bit overkill, but allows to catch an edge case
-      // where
-      // scheduleToStart is already fired, but didn't report a completion yet.
-      boolean shouldDiscardTheAttempt = scheduleToStartFired || executionContext.isCompleted();
-      if (shouldDiscardTheAttempt) {
-        return;
-      }
-
-      Scope metricsScope =
-          workerMetricsScope.tagged(
-              ImmutableMap.of(
-                  MetricsTag.ACTIVITY_TYPE,
-                  activityTask.getActivityType().getName(),
-                  MetricsTag.WORKFLOW_TYPE,
-                  activityTask.getWorkflowType().getName()));
-
-      MDC.put(LoggerTag.ACTIVITY_ID, activityTask.getActivityId());
-      MDC.put(LoggerTag.ACTIVITY_TYPE, activityTask.getActivityType().getName());
-      MDC.put(LoggerTag.WORKFLOW_ID, activityTask.getWorkflowExecution().getWorkflowId());
-      MDC.put(LoggerTag.WORKFLOW_TYPE, activityTask.getWorkflowType().getName());
-      MDC.put(LoggerTag.RUN_ID, activityTask.getWorkflowExecution().getRunId());
       try {
+        // if an activity was already completed by any mean like scheduleToClose or scheduleToStart,
+        // discard this attempt, this execution is completed.
+        // The scheduleToStartFired check here is a bit overkill, but allows to catch an edge case
+        // where scheduleToStart is already fired, but didn't report a completion yet.
+        boolean shouldDiscardTheAttempt = scheduleToStartFired || executionContext.isCompleted();
+        if (shouldDiscardTheAttempt) {
+          return;
```

Comment:
Here shouldn't the task slot release reason be not used?

Reply:
Not quite. This is a timeout, and the timeouts right now are also taskComplete. Not used is right now just "didn't even get a task"